src/sentry_transport.c

```diff
@@ -0,0 +1,115 @@
+#include "sentry_transport.h"
+#include "sentry_alloc.h"
+#include "sentry_slice.h"
+#include "sentry_utils.h"
+
+#define MAX_RATE_LIMITS 4
+
+struct sentry_rate_limiter_s {
+    uint64_t disabled_until[MAX_RATE_LIMITS];
+};
+
+sentry_rate_limiter_t *
+sentry__rate_limiter_new(void)
+{
+    sentry_rate_limiter_t *rl = SENTRY_MAKE(sentry_rate_limiter_t);
+    rl->disabled_until[SENTRY_RL_CATEGORY_ANY] = 0;
+    rl->disabled_until[SENTRY_RL_CATEGORY_ERROR] = 0;
+    rl->disabled_until[SENTRY_RL_CATEGORY_SESSION] = 0;
+    rl->disabled_until[SENTRY_RL_CATEGORY_TRANSACTION] = 0;
+    return rl;
+}
+
+bool
+sentry__rate_limiter_update_from_header(
+    sentry_rate_limiter_t *rl, const char *sentry_header)
+{
+    sentry_slice_t slice = sentry__slice_from_str(sentry_header);
+
+    while (true) {
+        slice = sentry__slice_trim(slice);
+        uint64_t retry_after = 0;
+        if (sentry__slice_pop_front_if(&slice, '+')) {
```

Comment:
This is gone now, see Python for reference: 
- https://github.com/getsentry/sentry-python/pull/651
- https://github.com/getsentry/sentry-python/pull/652

Reply:
So, is it `Rate-Limit` or `Rate-Limits`? :-D